lib/Conversion/StableHLOToTTIR/StableHLOToTTIRPatterns.cpp

```diff
@@ -133,19 +133,103 @@ class StableHLOToTTIRTransposeOpConversionPattern
 
-    assert(adaptor.getPermutation().size() == 2 &&
-           "TTIR only supports only two dimensional transposeOp.");
+    if (!checkBasicLegality(srcOp, adaptor)) {
+      return failure();
+    }
 
     rewriter.replaceOpWithNewOp<mlir::tt::ttir::TransposeOp>(
         srcOp, outputTensor.getType(), Value(adaptor.getOperand()),
         Value(outputTensor), adaptor.getPermutation()[0],
         adaptor.getPermutation()[1],
+
+        rewriter.getArrayAttr(
+            SmallVector<Attribute>(adaptor.getOperands().size() + 1,
+                                   rewriter.getAttr<OperandConstraintAttr>(
+                                       OperandConstraint::AnyDeviceTile))));
+    return success();
+  }
+
+  bool
+  checkBasicLegality(mlir::stablehlo::TransposeOp &srcOp,
+                     mlir::stablehlo::TransposeOp::Adaptor &adaptor) const {
+
+    if (adaptor.getPermutation().size() != 2) {
+      srcOp->emitError()
+          << "TTIR only supports only two dimensional transposeOp.";
+      return false;
+    }
+
+    return true;
+  }
+};
+
+class StableHLOToTTIRDotGeneralOpConversionPattern
+    : public OpConversionPattern<mlir::stablehlo::DotGeneralOp> {
+  using OpConversionPattern<mlir::stablehlo::DotGeneralOp>::OpConversionPattern;
+
+public:
+  LogicalResult
+  matchAndRewrite(mlir::stablehlo::DotGeneralOp srcOp,
+                  mlir::stablehlo::DotGeneralOp::Adaptor adaptor,
+                  ConversionPatternRewriter &rewriter) const override {
+    auto outputType = mlir::cast<RankedTensorType>(srcOp.getResult().getType());
+    tensor::EmptyOp outputTensor = rewriter.create<tensor::EmptyOp>(
+        srcOp.getLoc(), outputType.getShape(), outputType.getElementType());
+
+    // This is a basic version that can only work for cases that can be directly
+    // converted to matmul. The op should be extended as other ops such as
+    // ttir.permute and ttir.broadcast_in_dim become available.
+
+    if (!checkBasicLegality(srcOp, adaptor)) {
+      return failure();
+    }
+
+    rewriter.replaceOpWithNewOp<mlir::tt::ttir::MatmulOp>(
+        srcOp, outputTensor.getType(), adaptor.getLhs(), adaptor.getRhs(),
+        Value(outputTensor),
         rewriter.getArrayAttr(
             SmallVector<Attribute>(adaptor.getOperands().size() + 1,
                                    rewriter.getAttr<OperandConstraintAttr>(
                                        OperandConstraint::AnyDeviceTile))));
     return success();
   }
+
+private:
+  bool
+  checkBasicLegality(mlir::stablehlo::DotGeneralOp &srcOp,
+                     mlir::stablehlo::DotGeneralOp::Adaptor &adaptor) const {
+
+    ::mlir::stablehlo::DotDimensionNumbersAttr dimensions =
+        adaptor.getDotDimensionNumbers();
+
+    if (dimensions.getLhsContractingDimensions().size()) {
+      if (dimensions.getLhsContractingDimensions()[0] != 1) {
+        srcOp->emitError()
+            << "ttir dot_general only supports matmul operation.";
```

Comment:
TTIR doesn't have a `dot_general` function, I think you should change this message to something like:
```
Only regular non-transposed matmul is supported currently in TTIR.
```